CONTRIBUTING.md

```diff
@@ -34,9 +34,19 @@ The project aims to minimise the impact on users upgrading to newer versions of
 
 To minimise such disruption to users, where a change to the interface is required, all changes to the ABI or API, even on the main development branch where new features are added, need to be justifiable by either being a significant enhancement, new feature or bug fix which is best resolved by an interface change. If there is an API change, the contribution, if accepted, will be merged only when there will be a major release.
 
-Where changes to an existing interface are necessary, functions in the public interface which need to be changed, are marked as 'deprecated'. This is done with the preprocessor symbols `MBEDTLS_DEPRECATED_WARNING` and `MBEDTLS_DEPRECATED_REMOVED`. Then, a new function with a new name but similar if not identical behaviour to the original function containing the necessary changes should be created alongside the existing deprecated function.
-
-When a build is made with the deprecation preprocessor symbols defined, a compiler warning will be generated to warn a user that the function will be removed at some point in the future, notifying users that they should change from the older deprecated function to the newer function at their own convenience.
+Where changes to an existing interface are necessary, functions in the public interface which need to be changed are marked as 'deprecated'. Then, a new function with a new name but similar if not identical behaviour to the original function containing the necessary changes should be created alongside the existing deprecated function.
+
+To deprecate a function, the following actions must be taken:
+* In the header file (e.g. `include/mbedtls/xyz.h`):
+    * Surround the function declaration with `#if !defined(MBEDTLS_DEPRECATED_REMOVED)`.
+    * Add a line to the docstring of the form `\deprecated Reason for deprecation`.
```

Comment:
I remember a while ago we had a debate as to whether `\deprecated` should be used in cases where only one aspect of the behavior of a function is deprecated, but the function itself is staying. Things like “it is now deprecated to call this function with a negative argument”. I don't remember where this debate was (might not have been on GitHub unfortunately) or what the conclusion was.

Reply:
In 2020 we decided to use `\deprecated` in such cases, for example [wrong-way RSA](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-2.28.0/include/mbedtls/rsa.h#L594). But previously we'd gone the other way, for example [RSA private without blinding](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-2.28.0/include/mbedtls/rsa.h#L559).